pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
+            if it is a nested list, it is a ``list`` of ``lists``, they are different time set for different var in var_name
+        values: a ``list`` containing values which has the same shape of flattened variables 
+            default choice is None, means there is no give nvalues 
+        upper_bound: a ``list`` of upper bounds. If given a scalar number, it is set as the upper bounds for all variables.
+        lower_bound: a ``list `` of lower bounds. If given a scalar number, it is set as the lower bounds for all variables.
+
+        Return
+        ------
+        if not defining values, return a set of variable names 
+        if defining values, return a dictionary of variable names and its value 
         """
-        self.measurement_all_info = measurement_index_time
-        self.ind_string = ind_string
-        # a list of measurement names
-        self.measurement_name = list(measurement_index_time.keys())
-        # begin flatten
-        self._name_and_index_generator(self.measurement_all_info)
-        self._generate_flatten_name(self.name_and_index)
-        self._generate_variance(self.flatten_measure_name, variance, self.name_and_index)
-        self._generate_flatten_timeset(self.measurement_all_info, self.flatten_measure_name, self.name_and_index)
-        self._model_measure_name()
-
-        # generate the overall measurement time points set, including the measurement time for all measurements
-        flatten_timepoint = list(self.flatten_measure_timeset.values())
-        overall_time = []
-        for i in flatten_timepoint:
-            overall_time += i
-            timepoint_overall_set = list(set(overall_time))
-        self.timepoint_overall_set = timepoint_overall_set
-
-
-    def _name_and_index_generator(self, all_info):
+        self._names(var_name, extra_index,time_index, values, upper_bound, lower_bound, check_values_bounds=False)
+
+        self._add_elements(var_name, extra_index=extra_index, time_index=time_index)
+
+        self._names(var_name, extra_index,time_index, values, upper_bound, lower_bound, check_values_bounds=True)
+
+        if values:
+            # this dictionary keys are special set, values are its value
+            self.special_set_value = self._generate_dict(values)
+
+        if upper_bound:
+            if type(upper_bound) in [int, float]:
+                upper_bound = [upper_bound]*len(self.special_set)
+            self.upper_bound = self._generate_dict(upper_bound)
+            
+        if lower_bound:
+            if type(lower_bound) in [int, float]:
+                lower_bound = [lower_bound]*len(self.special_set)
+            self.lower_bound = self._generate_dict(lower_bound)
+
+        return self.special_set
+    
+    def update_values(self, values):
         """
-        Generate a dictionary, keys are the variable names, values are the indexes of this variable.
-        For e.g., name_and_index = {'C': ['CA', 'CB', 'CC']}
+        Used for updating values
```

Comment:
Every time we change an experiment, like in grid search, or for optimization starting from different initial point, we can change design variable values. The point is the design variable names are already generated and stored by `DesignVariables`, user only needs to provide  a list of values to update the values for these names. I think there is no need to change the way we do this, since no matter how many times user call `add_measurements` to add design variables, when they need to update design variable values, all names should have been defined and stored.